### DISCUSSION TITLE
Can only get env variables from "Base Environment" in pre-request script

### DISCUSSION THREAD

**Discussion by alaz-aura** [Kong/insomnia#7550] [Help] [answered]
if using a pre-request script, there doesn't seem a way to use `insomnia.<insertCustomEnvironmentNameHere>.get("variable")`

It does however only seem to work for `insomnia.baseEnvironment.get("variable")`
<img width="334" alt="Screenshot 2024-06-14 at 10 33 45 pm" src="https://github.com/Kong/insomnia/assets/144073715/737820ca-4ec3-4223-b378-248d621a0aa7">

attempting to use `insomnia.testing.get("variable")` produces:

`Cannot read properties of undefined (reading 'get')`
<img width="629" alt="Screenshot 2024-06-14 at 10 35 17 pm" src="https://github.com/Kong/insomnia/assets/144073715/2d88dfb0-708f-4747-9285-c281d282a4da">

**Comment by ihexxa** [comment, marked as answer]
Hello @alaz-aura , Insomnia supports to access the selected environment, for example:
- make sure that a "sub environment" (testing) is selected in the left top environment menu
- try `insomnia.environment.get('hello')` to read `hello from `testing`

**Reply by alaz-aura** [reply]
@ihexxa I've tried that many times, with many variations, however I still don't get a success. With `testing` being my custom environment, I've tried: 
* insomnia.environment.testing.get('hello')
* insomnia.baseEnvironment.testing.get('hello')
* insomnia.testing.get('hello')

The response is always: `Cannot read properties of undefined (reading 'get')`

**Reply by alaz-aura** [reply]
@ihexxa after playing around with a new shared environment, it finally worked for me. 

Unsure what was going on before. I have used the following to successfully get an env value from the `testing` environment:
`insomnia.environment.get("env")`

![Screenshot 2024-06-18 at 8 15 33 pm](https://github.com/Kong/insomnia/assets/144073715/dc158cf7-3510-4353-8835-94930732ee73)

**Reply by yellowvamp04** [reply]
@ihexxa is there other way to select sub environment in pre-request script rather than selecting it manually on the top left menu?

**Reply by ihexxa** [reply]
Hi @yellowvamp04 Nope, currently Insomnia exposes "selected" environment automatically. We will provide "folder-level env" interfaces which should work in this case: https://github.com/Kong/insomnia/pull/8348
